spyder/plugins/editor/widgets/tests/test_panels.py

```diff
@@ -8,16 +8,62 @@
 '''
 
 # Third party imports
-from qtpy.QtGui import QTextCursor
+from qtpy.QtCore import QSize, Qt
+from qtpy.QtGui import QTextCursor, QPainter, QColor, QFontMetrics
 import pytest
 
 # Local imports
+from spyder.api.panel import Panel
 from spyder.utils.qthelpers import qapplication
-from spyder.plugins.editor.widgets.codeeditor import CodeEditor
 from spyder.plugins.editor.panels.linenumber import LineNumberArea
 from spyder.plugins.editor.panels.edgeline import EdgeLine
 from spyder.plugins.editor.panels.scrollflag import ScrollFlagArea
 from spyder.plugins.editor.panels.indentationguides import IndentationGuide
+from spyder.plugins.editor.widgets.codeeditor import CodeEditor
+
+
+# --- External Example Panel
+# -----------------------------------------------------------------------------
+class EmojiPanel(Panel):
+    """Example external panel."""
+
+    def __init__(self):
+        """Initialize panel."""
+        Panel.__init__(self)
+        self.setMouseTracking(True)
+        self.scrollable = True
+
+    def sizeHint(self):
+        """Override Qt method.
+        Returns the widget size hint (based on the editor font size).
+        """
+        fm = QFontMetrics(self.editor.font())
+        size_hint = QSize(fm.height(), fm.height())
+        if size_hint.width() > 16:
+            size_hint.setWidth(16)
+        return size_hint
+
+    def _draw_red(self, top, painter):
+        """Draw emojis.
+        Args:
+            top (int): top of the line to draw the breakpoint icon.
+            painter (QPainter)
```

Comment:
I think you need to rewrite these lines because it seems you copied them from the debugger panel.